diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/WhiteSpaceRule.kt

```diff
@@ -61,14 +90,88 @@ class WhiteSpaceRule : Rule("horizontal-whitespace") {
 
     private fun handleKeywordWithParOrBrace(node: ASTNode) {
         if (node.treeNext.numWhiteSpaces() != 1) {
+            // there is either not single whitespace or newline after keyword
+            val nextCodeLeaf = node.nextCodeLeaf()!!
+            if (nextCodeLeaf.elementType != LPAR && nextCodeLeaf.elementType != LBRACE) {
+                // keywords without code block, e.g. else. Fixme: should be handled too
+                return
+            }
             WRONG_WHITESPACE.warnAndFix(configRules, emitWarn, isFixMode, "keyword '${node.text}' should be separated from " +
-                    "'${if (node.elementType in keywordsWithPar) "(" else "{"}' with a whitespace", node.startOffset) {
+                    "'${nextCodeLeaf.text}' with a whitespace", nextCodeLeaf.startOffset) {
                 node.leaveSingleWhiteSpace()
             }
         }
     }
 
-    private fun ASTNode.numWhiteSpaces() = if (elementType == WHITE_SPACE) text.count { it == ' ' } else 0
+    /**
+     * This method covers all other opening braces, not covered in [handleKeywordWithParOrBrace].
+     */
+    private fun handleLbrace(node: ASTNode) {
+        val whitespaceOrPrevNode = node.parent({ it.treePrev != null }, strict = false)!!.treePrev
+        val isFromLambdaAsArgument = node.parents().take(numParentsForLambda).toList().let {
+            it[0].elementType == FUNCTION_LITERAL &&
+                    it[1].elementType == LAMBDA_EXPRESSION &&
+                    it[2].elementType == VALUE_ARGUMENT
```

Comment:
don't you think that you can get index out of bound in some cases here?

Reply:
Right, I will add a check.